e2e/action-row.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { test, expect, type Page, type Locator } from '@playwright/test';
+
+const challengeButtons = [
+  'Instructions',
+  'index.html',
+  'styles.css',
+  'Console'
+];
+
+const editorButtons = ['index.html', 'styles.css'];
+
+let page: Page;
+
+test.beforeAll(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto(
+    '/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form'
+  );
+});
+
+test.afterAll(async () => {
+  await page.close();
+});
+
+function getActionRowLocator(): Locator {
+  return page.getByTestId('action-row');
+}
+
+function getTabsRowLocator(): Locator {
+  return page.getByTestId('action-row');
+}
+
+test('Action row buttons are visible', async ({ isMobile }) => {
```

Comment:
```suggestion
test('Action row buttons are visible', async ({ isMobile, page }) => {
```

We can get `page` directly from the function args.